configure.ac

```diff
@@ -33,6 +33,17 @@ if test "x$enable_debug" = "xyes" ; then
   CXXFLAGS="$CXXFLAGS -D__SOMDebugPrint__ -Dmylockdebug -DcoherenceCheck -D__SASDebugPrint__"
 fi
 
+# Hardware Transactional Memory option
+AC_ARG_ENABLE([htm],
+  [AS_HELP_STRING([--enable-htm],
+    [Enable Hardware Transactional Memory])],
+  [],
+  [enable_htm=no])
+
+HTM=""
+AS_IF([test "x$enable_htm" != xno],
+  [AC_SUBST([HTM], [-mcpu=power8])])
```

Comment:
> And that brings another question: what if an user wants to build SPHDE for a POWER7? This change will silently add an illegal instruction to his/her code.

No, it won't.  My mistake.
Anyway, you can still automatically enable HTM by using the solution I proposed or ignore it and let the user enable it only if specified.